spec.md

```diff
@@ -2786,8 +2878,9 @@ module = "module" , id , ":" , [ info ] , newline , indent ,
            { port , newline } ,
            { statement , newline } ,
          dedent ;
+port_extmodule = ( "input" | "output" ) , id , ":": , type_ground | type_aggregate , [ info ] ;
 extmodule = "extmodule" , id , ":" , [ info ] , newline , indent ,
-              { port , newline } ,
+              { port_extmodule , newline } ,
```

Comment:
@seldridge I think this is a bit messy, but you said you wanted to clean up the types later, so not sure if this flies. The notable thing that @mwachs5 and I discovered yesterday is that extmodules can't have uninferred `Reset`-type ports.

Reply:
Ok, if we are not talking about this here we should mention it in the extmodule section I guess.

Overall I would desire that we could have inferred Resets (both inputs and outputs) on extmodule since it simplifies things  -- the other rules should kick in to interpret and enforce the inference rules. Such as if I have a Reset coming out of a black box and it drives an AsyncReset, the other rules already say we should know that extmodule's output is AsyncReset

Reply:
ok great. Let's just put a bit in the `extmodule` section then about that?

Reply:
So, technically that test is wrong. SFC allows uninferred resets on external module inputs, but not outputs. The test should be split into one test of input and one test of output.

I am actually not sure that this restriction has to be made. I.e., the reset inference algorithm should always do the right thing here.

Reply:
hmmm does MFC support resets on extmodule ports? I guess we can leave this stuff out if so.

Reply:
For the following test:

Reply:
Cool, I'll just not say anything special about extmods then, since that makes the spec simpler, too!